fintorch/models/timeseries/causalformer/causalformer_module.py

```
@@ -0,0 +1,297 @@
+
+        # Handle target tensor shape - ensure it becomes 4D [batch_size, num_series, future_length, feature_dim]
+        if y.ndim == 2:  # [batch_size, future_length]
+            # Add both series and feature dimensions
+            y = y.unsqueeze(2).unsqueeze(-1)  # [batch_size, future_length, 1, 1]
+            y = y.permute(0, 2, 1, 3)  # [batch_size, 1, future_length, 1]
+        elif y.ndim == 3:  # [batch_size, future_length, features] - from multi-series processing
+            # Add series dimension and feature dimension if needed
+            if y.shape[-1] == 1:  # Already has feature dimension
+                y = y.unsqueeze(1)  # [batch_size, 1, future_length, 1]
+            else:  # Multiple features, add series dimension
+                y = y.unsqueeze(1)  # [batch_size, 1, future_length, features]
```

Ruff (0.11.9)

Use ternary operator `y = y.unsqueeze(1) if y.shape[-1] == 1 else y.unsqueeze(1)` instead of `if`-`else`-block

(SIM108)